lib/ansible/modules/windows/win_chocolatey.ps1

```diff
@@ -271,23 +271,21 @@ Function Install-Chocolatey {
         # locate the newly installed choco.exe
         $choco_app = Get-Command -Name choco.exe -CommandType Application -ErrorAction SilentlyContinue
         if ($null -eq $choco_app) {
-            $choco_path = $env:ChocolateyInstall
-            if ($null -ne $choco_path) {
```

Comment:
This may actually be the issue we had before, this should really be `if (-not (Test-Path -LiteralPath $choco_path))` and not checking for `$null`.

Reply:
That's what I first thought, but it's not the case. We only use `$choco_path` as an attempt to find `choco.exe`.
That's why I changed the logic so it's less deceptive (called it $choco_dir). We do a second lookup and we should be checking if `$choco_app` is `$null` before accepting the Path property.

But this is not the cause of the issue, the cause is that after installing Chocolatey `choco.exe` was not found. Maybe the installation failed silently? Next time when it happens (and it's repeatable) we need to investigate.

Reply:
I don't understand what this PR changes here except how the path is formed. Unless I'm missing something both the before and after does;

* Checks if `$env:ChocolateyInstall` exists, if so it  uses that like `$env:ChocolateyInstall\bin\choco.exe`
* If the above doesn't exist, use `$env:SYSTEMDRIVE\ProgramData\Chocolatey\bin\choco.exe`

There's also an issue where you've change the variables to be `$choco_dir` but you are still checking `$choco_path`.

Reply:
Yes, they are identical, but I don't like the before-code because it confused me. The after-code stores the same value in $choco_path which is then used identically. The before-code appended stuff to $choco_path afterwards, or replaces it. So it has in-flux data that does not correspond to what you expect.